cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -254,9 +256,21 @@ def validate_and_update_from_json(cls, descriptor, jsondict, user, filter_tabs=T
                 val = model['value']
                 if hasattr(descriptor, key) and getattr(descriptor, key) != val:
                     key_values[key] = descriptor.fields[key].from_json(val)
-            except (InvalidProctoringProvider, TypeError, ValueError, ValidationError) as err:
+            except (TypeError, ValueError, ValidationError) as err:
                 did_validate = False
                 errors.append({'key': key, 'message': str(err), 'model': model})
+            except InvalidProctoringProvider as err:
```

Comment:
I think no matter which way we choose to implement this, it's going to be a little gross. There's no way for the above `from_json` to get the course_id without making changes to each of the field models, so we have to check for the error again here to ensure that the correct error message is being included.